tests/gp_tests/test_acqf.py

```diff
@@ -64,3 +66,65 @@ def test_eval_acqf(
 
     assert torch.all(torch.isfinite(acqf_value))
     assert torch.all(torch.isfinite(acqf_grad))
+
+
+@pytest.mark.parametrize(
+    "x", [np.array([0.15, 0.12]), np.array([[0.15, 0.12], [0.0, 1.0]])]  # unbatched  # batched
+)
+@pytest.mark.parametrize(
+    "c",
+    [
+        [np.array([2.0, 3.5, 4.0])],
+        [np.array([2.0, 3.5, 4.0]), np.array([2.5, 1.8, 2.1])],
+    ],
+)
+def test_eval_acqf_with_constraints(x: np.ndarray, c: list[np.ndarray]) -> None:
+    n_dims = 2
+    X = np.array([[0.1, 0.2], [0.2, 0.3], [0.3, 0.1]])
+    Y = np.array([1.0, 2.0, 3.0])
+    kernel_params = KernelParamsTensor(
+        inverse_squared_lengthscales=torch.tensor([2.0, 3.0], dtype=torch.float64),
+        kernel_scale=torch.tensor(4.0, dtype=torch.float64),
+        noise_var=torch.tensor(0.1, dtype=torch.float64),
+    )
+    search_space = SearchSpace(
+        scale_types=np.full(n_dims, ScaleType.LINEAR),
+        bounds=np.array([[0.0, 1.0] * n_dims]),
+        steps=np.zeros(n_dims),
+    )
+
+    acqf_params = create_acqf_params(
+        acqf_type=AcquisitionFunctionType.LOG_EI,
+        kernel_params=kernel_params,
+        search_space=search_space,
+        X=X,
+        Y=Y,
+        acqf_stabilizing_noise=0.0,
+    )
+    constraints_acqf_params = [
+        create_acqf_params(
+            acqf_type=AcquisitionFunctionType.LOG_PI,
+            kernel_params=kernel_params,
+            search_space=search_space,
+            X=X,
+            Y=vals,
+            acqf_stabilizing_noise=0.0,
```

Comment:
Probably, you need to explicitly `max_Y` to this function so that you can test the following three cases:
1. Only feasible solutions
2. Only infeasible solutions
3. Both feasible and infeasible solutions